Sources/WebDriver/Session.swift

```diff
@@ -30,16 +31,34 @@ public class Session {
             owned: true)
     }
 
-    public static func createW3C(webDriver: any WebDriver, alwaysMatch: Capabilities, firstMatch: [Capabilities] = []) throws -> Session {
+    /// Initializer for W3C Protocol
+    private convenience init(webDriver: any WebDriver, alwaysMatch: Capabilities, firstMatch: [Capabilities]) throws {
         let response = try webDriver.send(Requests.Session_W3C(
             alwaysMatch: alwaysMatch, firstMatch: firstMatch))
-        return Session(
+        self.init(
             webDriver: webDriver,
             existingId: response.value.sessionId,
             capabilities: response.value.capabilities,
             owned: true)
     }
 
+    public convenience init(webDriver: any WebDriver, capabilities: Capabilities) throws {
+        switch webDriver.wireProtocol {
+            case .legacySelenium:
+                try self.init(webDriver: webDriver, desiredCapabilities: capabilities, requiredCapabilities: capabilities)
+            case .w3c:
+                try self.init(webDriver: webDriver, alwaysMatch: capabilities, firstMatch: [])
+        }
+    }
+
+    public static func createLegacy(webDriver: any WebDriver, desiredCapabilities: Capabilities, requiredCapabilities: Capabilities? = nil) throws -> Session {
```

Comment:
Avoid needless words - consider `static func legacy(driver:preferredCapabilities:requiredCapabilities:)`

Reply:
I ended up reusing the enum scoping idea: `Session.LegacySelenium.create(...)`